### ISSUE TITLE
Stickup cams not capable of motion detection?

### ISSUE THREAD

**Issue by BillyCroan** [tchellomello/python-ring-doorbell#303]
I have a ring doorbell at the front door, and a stickup cam above the driveway.

Both detect motion and can chime indoors and on our phones via the app.  Yet this software states that the stickup cam is incapable of motion detection.  

ring-mqtt can set motion detection on or off for the stickup cam.  I was looking for this project to be able to do the same more quickly, directly called by existing python code.  But I think there is a bug if it's saying that stickup cams don't support motion detection.

```
root@RingPi:~# ring/bin/ring-doorbell motion-detection --device-name "Front Door"  --off
---------------------------------
Ring CLI
Front Door (cocoa_doorbell) motion detection set to off
root@RingPi:~# ring/bin/ring-doorbell motion-detection --device-name "Driveway"  --off
---------------------------------
Ring CLI
Driveway is not capable of motion detection
root@RingPi:~# ring/bin/ring-doorbell motion-detection --device-name "Driveway"  --on
---------------------------------
Ring CLI
Driveway is not capable of motion detection
root@RingPi:~# ring/bin/ring-doorbell 
---------------------------------
Ring CLI
Name:       Front Door
Family:     doorbots
ID:         360987170
Timezone:   America/Chicago
Wifi Name:  522fbr2
Wifi RSSI:  -43

Name:       Driveway
Family:     stickup_cams
ID:         2979156
Timezone:   America/Chicago
Wifi Name:  522g2
Wifi RSSI:  -30

Name:       Bedroom
Family:     chimes
ID:         551857
Timezone:   America/Chicago
Wifi Name:  522d2
Wifi RSSI:  -45

root@RingPi:~# 

```

**Comment by sdb9696** [comment]
Hi @BillyCroan, can you try this with the latest version of the library and if it still doesn't work tell me what the model is?

**Comment by rcanetta** [comment]
Hi
I have one StickUp Cam (3rd Gen) and two Spotlight Cam Pro, which ring-doorbell shows as "stickup_cams": on the first one the motion-detection switch works, on the other 2 it doesn't with same error above ("not capable"). I just updated to 0.8.2... any hint?

Thanks
Riccardo

**Comment by BillyCroan** [comment]
@sdb9696 I think it is still not working:

Installing collected packages: pytz, urllib3, sniffio, oauthlib, idna, charset-normalizer, certifi, asyncclick, requests, anyio, requests-oauthlib, ring_doorbell
Successfully installed anyio-4.1.0 asyncclick-8.1.3.4 certifi-2023.11.17 charset-normalizer-3.3.2 idna-3.6 oauthlib-3.2.2 pytz-2023.3.post1 requests-2.31.0 requests-oauthlib-1.3.1 ring_doorbell-0.8.2 sniffio-1.3.0 urllib3-2.1.0
root@RingPi:~# ring_doorbell_venv/bin/ring-doorbell motion-detection --device-name "Driveway" --off
---------------------------------
Ring CLI
Driveway is not capable of motion detection
root@RingPi:~# ring_doorbell_venv/bin/ring-doorbell motion-detection --device-name "Driveway" --on
---------------------------------
Ring CLI
Driveway is not capable of motion detection
root@RingPi:~# ring_doorbell_venv/bin/ring-doorbell motion-detection --device-name "Front Door"  --off
---------------------------------
Ring CLI
Front Door (cocoa_doorbell) motion detection set to off
root@RingPi:~# ring_doorbell_venv/bin/ring-doorbell motion-detection --device-name "Front Door"  --on
---------------------------------
Ring CLI
Front Door (cocoa_doorbell) motion detection set to on
root@RingPi:~# 
root@RingPi:~# ring_doorbell_venv/bin/ring-doorbell list
---------------------------------
Ring CLI
Front Door (cocoa_doorbell)
Bedroom (chime)
Driveway (stickup_cam_v3)
root@RingPi:~# 

from devices -dn Driveway:
Ring CLI
(Pretty format coming soon, if you want json consistently from this command provide the --json flag)
{
  "id": 2979156,
  "kind": "stickup_cam_v3",
  "description": "Driveway",
  "location_id": "9yutvj-4ive-0",
  "schema_id": null,
  "is_sidewalk_gateway": false,
  "created_at": "2017-07-08T13:23:53Z",
  "deactivated_at": null,
  "owner": {
    "id": 150510,
    "first_name": "",
    "last_name": "",
    "email": ""
  },
  "device_id": "9884e3d31040",
  "time_zone": "America/Chicago",
  "firmware_version": "Up to Date",
  "latitude": redacted,
  "longitude": redacted,
  "address": "",
  "owned": true,
  "stolen": false,
  "ring_id": null,
  "shared_at": null,
  "health": {
    "ac_power": 1,
    "battery_percentage": 6,
    "battery_percentage_category": "very_poor",
    "battery_save": true,
    "battery_voltage": 3710.0,
    "battery_voltage_category": "good",
    "connected": true,
    "device_type": "stickup_cam_v3",
    "external_connection": true,
    "firmware_version": "cam-12.0.38",
    "floodlight_on": false,
    "last_update_time": 1700931303,
    "network_connection_value": "wifi",
    "night_light_state": false,
    "night_mode_on": false,
    "ota_status": "timeout",
    "rss_connected": true,
    "rssi": -31.0,
    "rssi_category": "good",
    "second_battery_percentage_category": "unknown",
    "second_battery_voltage_category": "unknown",
    "sidewalk_connection": false,
    "white_led_on": false,
    "firmware_version_status": "Up to Date",
    "packet_loss": 0.0,
    "packet_loss_category": "good",
    "hatch_open": false,
    "ext_power_state": 17,
    "bandwidth": 4800,
    "status_time": 55735845958100,
    "vod_enabled": true,
    "battery_error": false,
    "bandwidth_last_time": 1690274996,
    "firmware_avg_bitrate": "994",
    "stream_resolution": 100,
    "video_packets_total": "1078",
    "wifi_is_ring_network": false,
    "wifi_name": "redacted",
    "battery_level": "lowest"
  },
  "subscribed": false,
  "subscribed_motions": false,
  "external_connection": true,
  "ring_net_id": null,
  "active_schedule_uuid": null,
  "alerts": {
    "connection": "online",
    "ota_status": "timeout",
    "battery": "lowest"
  },
  "motion_snooze": null,
  "battery_life": "6",
  "settings": {
    "motion_snooze_preset_profile": "none",
    "motion_snooze_presets": [
      "none",
      "low",
      "medium",
      "high"
    ],
    "live_view_preset_profile": "middle",
    "live_view_presets": [
      "low",
      "middle",
      "high",
      "highest"
    ],
    "doorbell_volume": 8,
    "advanced_motion_detection_enabled": true,
    "advanced_motion_detection_human_only_mode": false,
    "enable_audio_recording": true,
    "people_detection_eligible": true,
    "live_view_disabled": false,
    "ignore_zones": {
      redacted
    },
    "enable_rich_notifications": true,
    "rich_notifications_billing_eligible": false,
    "rich_notifications_face_crop_enabled": false,
    "loitering_threshold": 10,
    "advanced_motion_detection_types": [],
    "motion_detection_enabled": true,
    "rich_notifications_scene_source": "cloud",
    "offline_motion_event_settings": {
      "subscribed": false,
      "enabled": false,
      "max_upload_kb": 5000,
      "resolution_p": 360,
      "frequency_after_secs": 2,
        "period_after_secs": 30
    },
    "cv_settings": {
      "detection_types": {
        "human": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        },
        "loitering": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        },
        "motion": {
          "enabled": true,
          "mode": "edge",
          "record": null,
          "notification": true
        },
        "moving_vehicle": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        },
        "other_motion": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        },
        "package_delivery": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        },
        "package_pickup": {
          "enabled": false,
          "mode": "none",
          "record": null,
          "notification": false
        }
      },
      "threshold": {
        "loitering": 10.0,
        "package_delivery": 2.0
      },
      "triggers": []
    },
    "sheila_settings": {
      "cv_processing_enabled": false,
      "local_storage_enabled": false
    },
    "server_settings": {
      "ring_media_server_enabled": true,
      "ring_media_server_host": "rms-us-west-2.rapi.us-east-1.prod.client.cap.ring.devices.a2z.com"
    },
    "cv_paid_features": {
      "baby_cry": false,
      "car_alarm": false,
      "co2_smoke_alarm": false,
      "dog_bark": false,
      "general_sound": false,
      "glass_break": false,
      "human": true,
      "loitering": false,
      "motion": false,
      "moving_vehicle": false,
      "other_motion": false,
      "package_delivery": false,
      "package_pickup": false,
      "cv_triggers": false
    },
    "other_paid_features": {
      "alexa_concierge": false,
      "sheila_cv": false,
      "sheila_recording": false,
     "critical_alerts": false,
      "system_level_pip": false,
      "extended_live_view": false,
      "snapshot_capture_plus": null,
      "live_speak": null,
      "closed_caption": null,
      "daily_digest": null,
      "ding_call": null,
      "recording_24x7": null
    },
    "zone_settings": {
      "motion": [

          skipped
    },
    "stark_enrolled": null,
    "network_settings": {
      "mac_address_ble": null,
     "max_dynamic_listen_interval": null,
      "network_diagnosis": {
        "channel_analysis": null,
        "periodic_diagnosis": null,
        "lan_performance": null,
        "tcp_iperf": null,
        "tcp_host": null,
        "tcp_port": null,
        "udp_iperf": null,
        "udp_host": null,
        "udp_port": null,
        "udp_bandwidth": null
      },
      "multi_net_pref": 3,
      "mac_address_wifi_24": null,
      "mac_address_wifi_5": null
    },
    "motion_zones": [
      1,
      1,
      1,
      1,
      1
    ],
    "enable_vod": 1,
    "exposure_control": 2,
    "vod_suspended": 1,
    "vod_status": "suspended",
    "voice_volume": null,
    "video_settings": {
      "encryption_enabled": false,
      "encryption_method": 0,
      "recording_24x7_mode": null
    },
    "advanced_motion_zones": {
      "zone1": {
        "name": "Default Zone",
        "state": 2,
    "advanced_motion_zones_enabled": true,
    "advanced_motion_zones_type": "8vertices",
    "advanced_pir_motion_zones": {
      "zone1_sensitivity": 5,
      "zone2_sensitivity": 5,
      "zone3_sensitivity": 5,
      "zone4_sensitivity": 5,
      "zone5_sensitivity": 5,
      "zone6_sensitivity": 5
    },
    "pir_sensitivity_1": 2,
    "lite_24x7": {
      "subscribed": false,
      "enabled": false,
      "frequency_secs": 3600,
      "resolution_p": 360
    }
  },
  "ext_power_state": 17,
  "features": {
    "cfes_eligible": false,
    "motion_zone_recommendation": false,
    "motions_enabled": true,
    "show_recordings": false,
    "show_vod_settings": true,
    "rich_notifications_eligible": true,
    "show_offline_motion_events": false,
    "sheila_camera_eligible": true,
    "sheila_camera_processing_eligible": true,
    "dynamic_network_switching_eligible": false,
    "show_24x7_lite": true,
    "recording_24x7_eligible": null
  }

Does that tell you enough about my model to help, or do I need to get up on a ladder?  I'm half afraid that the plastic has gotten brittle since I put it up that touching it could break the mount but maybe i can photograph the stickers in situ.  Just let me know if that's needed.  I might also have photos of it from install lurking around somewhere.

**Comment by rcanetta** [comment]
Hi, just tried with 0.8.3 but same result "not capable of motion detection"...

Thanks
Riccardo

**Comment by sdb9696** [comment]
> Does that tell you enough about my model to help, or do I need to get up on a ladder? I'm half afraid that the plastic has gotten brittle since I put it up that touching it could break the mount but maybe i can photograph the stickers in situ. Just let me know if that's needed. I might also have photos of it from install lurking around somewhere.

@BillyCroan thats very helpful thanks. I’ll take a look on Monday. You might want to edit your message to delete the json output as there’s a few details in there you may not want on the web. 

It might help if you could look at the output for the 2 other devices that work and tell me whether they are also stickup_cam_v3. You can email it to me directly if it’s easier. 

No need for getting up a ladder please. 😀

**Comment by BillyCroan** [comment]
I know my other ring devices are an indoor chime and a doorbell. I only bought one ring stickup cam.  Is there some chance they might report as "stickup_cam_v3"?

**Comment by sdb9696** [comment]
Hi both, that output was very helpful thanks.  @rcanetta you have a newer type of camera that was not configured for ring-doorbell.  @BillyCroan it seemed the motion-detection config for your model was missing.  I've addressed both of these in 0.8.4.  Could you try it out and let me know please?

**Comment by rcanetta** [comment]
Hi @sdb9696 just tried and it works, fantastic!! 

Thanks a lot!!
R
